### PR TITLE
allow blt::cuda as target name for cuda

### PR DESCRIPTION
**Pull request by trws**
What it says on the tin. This is the minimum I could figure out to allow blt::cuda to mirror blt::hip.

### CODE REVIEW

**Review by davidbeckingsale — APPROVED** [review]
Looks reasonable. Do you also want to make the change for cuda_runtime?

**Comment by rhornung67** [comment]
This change gives the same CMake error in RAJA when moving to the latest desul. I think we should continue to live with the inconsistent namespacing between CUDA and HIP in BLT, until the `blt::*` namespacing is fully fleshed out in BLT and available in a BLT release.

**Comment by trws** [comment]
I tested it against the latest desul, and got no errors.  How did you test it @rhornung67?  If that's still broken there's no point, but it shouldn't be.

Happy to add one for cuda_runtime.

**Comment by trws** [comment]
Actually, the tests on desul/desul#117 include this update, and are currently passing.

**Comment by rhornung67** [comment]
@trws just looking at your RAJA PR https://github.com/LLNL/RAJA/pull/1563

CMake error for desul atomics build on lassen.

**Comment by trws** [comment]
I just tested it, it works when not run with uberenv.  My current guess is that something in that stack is overriding the BLT being used so we get an old one.  Will probably have to dig to find it.  Here are the outputs for the clang and gcc versions, the one on the left _should_ match what's in CI if it were using the same BLT:

image

File: cmake/thirdparty/BLTSetupCUDA.cmake [hunk 1]
```diff
@@ -173,3 +175,4 @@ blt_import_library(NAME       cuda_runtime
                    INCLUDES   ${CUDA_INCLUDE_DIRS}
                    LIBRARIES  ${CUDA_LIBRARIES}
                    EXPORTABLE ${BLT_EXPORT_THIRDPARTY})
+add_library(blt::cuda_runtime ALIAS cuda)
```
Comment (rhornung67) [inline cmake/thirdparty/BLTSetupCUDA.cmake:178]:
```suggestion
add_library(blt::cuda_runtime ALIAS cuda_runtime)
```

Should it be this instead?

**Comment by trws** [comment]
It most definitely should, thanks @rhornung67.

**Comment by rhornung67** [comment]
> It most definitely should, thanks @rhornung67.

I think I'm hearing the Benny Hill theme song in my head....

**Comment by white238** [comment]
I am going to merge this then immediately do a PR with namespacing the rest of the targets along with updating all documentation. I am planning on doing a breaking release for BLT this year and will remove the generic names at that point.

**Pull request merged**